Some cleanup to ease installation process.

vdaghan opened the pull request:
From now on we do not need to install the original game for setup to work. This should somewhat support portability.

vdaghan committed:
Some cleanup to ease installation process.

From now on we do not need to install the original game for setup to work. This should somewhat support portability.

Danjb1 commented on Open-Rival/include/Image.h:
```diff
@@ -4,6 +4,7 @@
 #include <fstream>
 #include <string>
 #include <vector>
+#include <memory>
```
Can we keep these alphabetised?

The way I've been organising the includes is:

```
pch
"This" header file (for the current .cpp file)

Library / system headers

Open-Rival headers
```

Danjb1 commented on setup/src/setup.cpp:
```diff
@@ -18,7 +19,7 @@ namespace fs = std::filesystem;
     for (const fs::directory_entry& entry : fs::directory_iterator(videoDir)) {
         const fs::path path = entry.path();
-        std::filesystem::copy(path, outputDir + "\\" + path.filename().string());
+        std::filesystem::copy(path, outputDir + "\\" + path.filename().string(), std::filesystem::copy_options::update_existing );
```
Can we split this over multiple lines?

I try to keep to an 80-character limit where possible.

    std::filesystem::copy(path, outputDir + "\\" + path.filename().string(),
        std::filesystem::copy_options::update_existing );

Danjb1 commented on setup/src/setup.cpp:
```diff
@@ -30,21 +31,39 @@ void copyVideos(std::wstring videoDir, std::string outputDir) {
-    try {
-        gameDir = Registry::RegGetString(
+    bool findDirectoryFromRegistry = true;
+    for ( int count{ 1 }; count < argc; ++count ) {
```
`int count{ 1 }` - what is this syntax?

Danjb1 commented on setup/src/setup.cpp:
```diff
@@ -30,21 +31,39 @@ void copyVideos(std::wstring videoDir, std::string outputDir) {
+            std::cout << "-d {directory}: loads all files from {directory}" << '\n';
+            return 0;
+        }
+        if ( 0 == strcmp( "-d", argv[count] ) and count + 1 <= argc ) {
```
I've used `&&` elsewhere - in fact I didn't even know `and` was a valid keyword until now!

Danjb1 commented on setup/src/setup.cpp:
```diff
@@ -30,21 +31,39 @@ void copyVideos(std::wstring videoDir, std::string outputDir) {
+            return 0;
+        }
+        if ( 0 == strcmp( "-d", argv[count] ) and count + 1 <= argc ) {
+            std::string tmpStr( argv[count + 1] );
+            gameDir = std::wstring( tmpStr.begin(), tmpStr.end() );
```
Is this just taking the argument and converting it to a unicode string?

Danjb1 reviewed (dismissed):
This looks great, thanks again for all your help!

My comments are mostly minor formatting suggestions. I guess your MSVC is probably auto-formatting the code in some ways. I don't have my heart set on any particular formatting rules, but I would like to keep things consistent at least.

vdaghan committed:
Some cleanup to ease installation process.

From now on we do not need to install the original game for setup to work. This should somewhat support portability.
